README.md

```diff
@@ -160,6 +160,109 @@ realm.writeBlocking {
 }
 ```
 
+## Observing data changes
+
+Realm support asynchronous observers on all its data structures.
+
+### Realm
+
+A Realm can be observed globally for changes on its data.
+
+```Kotlin
+realm.asFlow()
+  .collect { realmChange: RealmChange<Realm> ->
+      when(realmChange) {
+         is InitialRealm -> notifyRealmInitialized(realmChange.realm)
+         is UpdatedRealm -> notifyRealmUpdated(realmChange.realm)
+      }
+  }
+```
+
+### RealmObject
+
+Realm objects can be observed individually. A list of the changed field names is provided on each update.
+
+```Kotlin
+val person = realm.query<Person>("name = $0", "Carlo").find().first()
+
+person.collect { objectChange: ObjectChange<Person> ->
+      when(objectChange) {
+         is InitialObject -> setObjectUI(objectChange.obj)
+         is UpdatedObject -> {
+             updateObjectUI(
+                 objectChange.obj,
+                 objectChange.changedFields
+            )
+         }
+         is DeletedObject -> deleteObjectUI()
+      }
+  }
+```
+
+### RealmLists
+
+Realm data structures can be observed too. On `RealmList` on each update you receive what positions were inserted, changed or deleted.
+
+```Kotlin
+person.addresses.asFlow()
+  .collect { listChange: ListChange<Address> ->
+      when(listChange) {
+         is InitialList -> setListUI(listChange.list)
+         is UpdatedList -> {
+             updateListUI(
+                 listChange.list,
+                 listChange.deletionRanges,
+                 listChange.insertionRanges,
+                 listChange.changeRanges
+            )
+         }
+         is DeletedList -> deleteListUI()
+      }
+  }
+```
+
+### RealmResults
+
+Query results are also observable, and like `RealmList` on each update the inserted, changed and deleted indices are also provided.
+
+```Kotlin
+realm.query<Person>().find().asFlow()
```

Comment:
```suggestion
realm.query<Person>().asFlow()
```

I would rename this section to `RealmQuery` ... while it is true that you can observe changes to RealmResults as well, that shouldn't be the majority of use cases...calling `find()` executes the query on the main thread the first time, which is something we shouldn't promote.

Reply:
You are right, I was thinking on using a `RealmResults` not using the right pattern.